aea/aea_builder.py

```diff
@@ -1385,6 +1388,27 @@ def _load_and_add_components(
                 component = load_component_from_config(configuration, **kwargs)
                 resources.add_component(component)
 
+    def _populate_contract_registry(self):
+        """Populate contract registry."""
+        for configuration in self._package_dependency_manager.get_components_by_type(
+            ComponentType.CONTRACT
+        ).values():
+            configuration = cast(ContractConfig, configuration)
+
+            # load contract interface
+            path = Path(
+                configuration.directory, configuration.path_to_contract_interface
+            )
+            with open(path, "r") as interface_file:
+                contract_interface = json.load(interface_file)
+
+            contract_registry.register(
+                id=str(configuration.public_id),
+                entry_point=f"{configuration.prefix_import_path}.contract:{configuration.class_name}",
+                contract_config=configuration,
+                contract_interface=contract_interface,
+            )
```

Comment:
since we are going towards a stateless contract, it would make sense to avoid passing the configuration instance to it. Maybe just keep the `config` dictionary from the configuration object.